v2/car.go

```diff
@@ -47,11 +47,23 @@ type (
 
 // WriteTo writes this characteristics to the given writer.
 func (c Characteristics) WriteTo(w io.Writer) (n int64, err error) {
-	if err = writeUint64To(w, c.Hi); err != nil {
+	if err = binary.Write(w, binary.LittleEndian, c.Hi); err != nil {
+		return
+	}
+	n += uint64BytesSize
+	if err = binary.Write(w, binary.LittleEndian, c.Lo); err != nil {
+		return
+	}
+	n += uint64BytesSize
+	return
```

Comment:
Once you have more than a handful of lines I'd personally avoid naked returns: https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters

They're not wrong per se, but once a function goes past a trivial line length, it requires more work from the reader to remember what the implicit return parameters are (or if there were any!) and how they were assigned to.

Reply:
Will get this resolved when optimising Write in future PRs; thank you for pointing this out 🍻